src/hls-adapter.js

```diff
@@ -943,11 +1030,19 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
       switch (errorType) {
         case Hlsjs.ErrorTypes.NETWORK_ERROR:
           {
-            const code = this._requestFilterError ? Error.Code.REQUEST_FILTER_ERROR : Error.Code.HTTP_ERROR;
+            let code;
+            if (this._requestFilterError) {
+              code = Error.Code.REQUEST_FILTER_ERROR;
+            } else if (this._responseFilterError) {
+              code = Error.Code.RESPONSE_FILTER_ERROR;
+            } else {
+              code = Error.Code.HTTP_ERROR;
+            }
             if (
               [Hlsjs.ErrorDetails.MANIFEST_LOAD_ERROR, Hlsjs.ErrorDetails.MANIFEST_LOAD_TIMEOUT].includes(errorName) &&
               !this._triedReloadWithRedirect &&
-              !this._config.forceRedirectExternalStreams
+              !this._config.forceRedirectExternalStreams &&
+              !(this._requestFilterError || this._responseFilterError)
```

Comment:
i think maybe it is easier to read it like this:
!this._requestFilterError && !this._responseFilterError, no?